experimental/ComplexReflectionGroups/src/hermitian_things.jl

```diff
@@ -0,0 +1,81 @@
+# This file implements some basics to work with the Hermitian inner product over
+# number fields.
+
+# Ulrich Thiel, 2024
+
+export is_unitary
+export is_orthogonal
+
+
+###########################################################################################
+# Complex conjugation
+###########################################################################################
+# In Hecke v0.30.2 complex_conjugation was extended to more fields following my suggestion.
+# I will only do one minor extension following Tommy Hofmanns suggestion:
+complex_conjugation(K::QQAbField) = conj
+
+###########################################################################################
+# Hermitian scalar product
+###########################################################################################
+function scalar_product(v::AbstractAlgebra.Generic.FreeModuleElem{T}, w::AbstractAlgebra.Generic.FreeModuleElem{T}) where T <: QQAlgFieldElem
```

Comment:
No, and for exactly the reason you say: I need to involve complex conjugation on the field.

Reply:
First off, to avoid misunderstandings, let me emphasize that I am OK with merging this as it is for now.

But I still would like to clear this up on the long run. Perhaps it is already clear for @thofma and @fieker but I am not familiar with the background, so I am asking / pointing out the following:

First off, `dot` as defined in Julia, does involve complex conjugation:

> Compute the dot product between two vectors. For complex vectors, the first vector is conjugated.

However, it doesn't work on `AbstractAlgebra.Generic.FreeModuleElem`. For starters iteration over these is not implemented. Nor is `conj` implemented for most of our types. Which makes me wonder: why not? We could just add
```julia
Base.conj(x::ZZRingElem) = x
Base.conj(x::ZZPolyRing) = x
Base.conj(x::ZZMPolyRing) = x

Base.conj(x::QQFieldElem) = x
Base.conj(x::QQPolyRing) = x
Base.conj(x::QQMPolyRing) = x

...
```

Reply:
`dot` implies conjugation on the first vector, where as @ulthiel wants it on the second vector. We use the same convention as @ulthiel for `inner_product`.

I think adding `conj` for ring elements, which are contained in the complex numbers is fine. Although I am not sure I want to extend it automatically to polynomials.